server/server.go

```diff
@@ -0,0 +1,359 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"go_todo_final/config"
+	"go_todo_final/model"
+	"go_todo_final/services/todolist"
+	"go_todo_final/transform"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+const (
+	webDir = "./web"
+
+	nowParam    = "now"
+	dateParam   = "date"
+	idParam     = "id"
+	repeatParam = "repeat"
+	searchParam = "search"
+)
+
+type TodoList interface {
+	AddTask(task model.Task) (int64, error)
+	GetTasks() ([]model.Task, error)
+	GetTasksByDate(date string) ([]model.Task, error)
+	GetTasksByString(searchFor string) ([]model.Task, error)
+	GetTaskById(id uint64) (model.Task, error)
+	UpdateTask(task model.Task) error
+	TaskDone(id int) error
+	DeleteTask(id int) error
+}
+
+type Server struct {
+	httpServer *http.Server
+	todoList   TodoList
+}
+
+func New(cfg *config.Config, todoList TodoList) *Server {
+	s := &Server{
+		httpServer: &http.Server{
+			Addr:              cfg.Port,
+			ReadHeaderTimeout: 1 * time.Second,
+		},
+		todoList: todoList,
+	}
+
+	s.setupHandlers()
+
+	return s
+}
+
+func (s *Server) Start() error {
+	return s.httpServer.ListenAndServe()
+}
+
+func (s *Server) setupHandlers() {
+	http.Handle("/", http.FileServer(http.Dir(webDir)))
+
+	http.HandleFunc("POST /api/signin", loginHandler)
+
+	http.HandleFunc("GET /api/nextdate", s.nextDateHandler)
+	http.HandleFunc("POST /api/task/done", auth(s.taskDoneHandler))
+	http.HandleFunc("DELETE /api/task", auth(s.deleteTaskHandler))
+	http.HandleFunc("POST /api/task", auth(s.addTaskHandler))
+	http.HandleFunc("GET /api/task", auth(s.getTaskByIdHandler))
+	http.HandleFunc("PUT /api/task", auth(s.updateTaskHandler))
+	http.HandleFunc("GET /api/tasks", auth(s.getTasksHandler))
+}
+
+func (s *Server) Stop(ctx context.Context) error {
+	return s.httpServer.Shutdown(ctx)
+}
+
+func (s *Server) nextDateHandler(w http.ResponseWriter, r *http.Request) {
+	date, err := time.Parse("20060102", r.URL.Query().Get(dateParam))
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+
+		return
+	}
+
+	now, err := time.Parse("20060102", r.URL.Query().Get(nowParam))
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+
+		return
+	}
+
+	repeat := r.URL.Query().Get(repeatParam)
+	nextDate, err := todolist.NextDate(date, now, repeat)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+
+		return
+	}
+
+	w.Write([]byte(nextDate.Format("20060102")))
+}
+
+func (s *Server) addTaskHandler(w http.ResponseWriter, r *http.Request) {
+	var incomingTask model.TaskDTO
+
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+
+	if err := json.NewDecoder(r.Body).Decode(&incomingTask); err != nil {
+		http.Error(w, fmt.Sprintf(`{"error":"%s"}`, err.Error()), http.StatusBadRequest)
```

Comment:
Так тоже будет работать

```suggestion
		http.Error(w, fmt.Sprintf(`{"error":"%v"}`, err), http.StatusBadRequest)
```